x-pack/legacy/plugins/ml/public/application/data_frame_analytics/common/analytics.ts

```diff
@@ -261,7 +261,7 @@ export function getEvalQueryBody({
 
   if (searchQuery !== undefined && ignoreDefaultQuery === true) {
     query = searchQuery;
-  } else if (isRegressionResultsSearchBoolQuery(searchQuery)) {
+  } else if (searchQuery !== undefined && isRegressionResultsSearchBoolQuery(searchQuery)) {
```

Comment:
both checks contain `searchQuery !== undefined`, maybe it worths to wrap in dedicated `if` statement

Reply:
I think for this case as it's only two conditions it feels more clean to avoid nested if statements.

Reply:
```suggestion
  let query: RegressionResultsSearchQuery = {
    term: { [`${resultsField}.is_training`]: { value: isTraining } },
  };
  
  if (searchQuery === undefined) {
    return query;
  }

  if (ignoreDefaultQuery === true) {
    query = searchQuery;
  } else if (isRegressionResultsSearchBoolQuery(searchQuery)) {
    const searchQueryClone = cloneDeep(searchQuery);
    searchQueryClone.bool.must.push(query);
    query = searchQueryClone;
  }
  return query;
```